### PR TITLE
Backend contests and Master changes

### PR DESCRIPTION
**Pull request by parthvshah**
Contains updates from upstream/master and added functionality in contrests. Specifics:
The following are from upstream/master
1) JWT
2) HTTPS at 8443
3) Fix verifyUser
The others are related to feature/contests. Parking this feature for now. Different updates separated by commits.

### CODE REVIEW

**Comment by aniketnk** [comment]
@parthvshah, next time make the merge from master commit separate. Don't need a PR for that.

**Comment by aniketnk** [comment]
Code looks good to me. @adityavinodk, check it out and merge.

File: server/routes/api/contests.js [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        });
+      })
+    }), // Works
+
+    app.post('/api/contests/updateContender', requireRole("admin"), function (req, res) {
+      var email = req.body.email;
+      var contestName = req.body.contestName;
+      var score = req.body.score;
+      var rank = req.body.rank;
+      var currentRank = req.body.currentRank;
```
Comment (adityavinodk) [inline server/routes/api/contests.js]: What is the difference between rank and currentRank?

File: server/routes/api/contests.js [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+          if (contests.length != 1) {
+            return res.status(404).send({
+              success: false,
+              message: 'Error: Contest not found.'
+            });
+          }
+          var contest = contests[0].toObject();
+          newHistory = {"contest": contest, "score": score, "rank": rank};
+          Contender.findOneAndUpdate({user: user}, {$push: {history: newHistory}});
+          Contender.findOneAndUpdate({user: user}, {currentRank: currentRank});
```
[the remark this answers is not in the document]
Comment (parthvshah) [reply server/routes/api/contests.js]: Okay. Will do.

File: server/routes/api/contests.js (hunk 1 above)
Comment (parthvshah) [reply server/routes/api/contests.js]: Rank is for the rank as part of history. It is the rank for that contest. currentRank is the contender's overall rank in Alcoding itself.

File: server/routes/api/contests.js [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+      });
+
+        return res.status(200).send({
+          success: true,
+          message: 'Handle Added succesfully',
+        });
+      })
+    }), // Works
+
+    app.post('/api/contests/updateContender', requireRole("admin"), function (req, res) {
```
[the remark this answers is not in the document]
Comment (parthvshah) [reply server/routes/api/contests.js]: Like you said, it's to update any fields you wish to of the contender.

**Comment by aniketnk** [comment]
[Contests Backend API framework](https://trello.com/c/vJ0OPXK4/47-contests-backend-api-framework)

**Pull request merged**